src/main/scala/uk/gov/nationalarchives/tdr/api/service/FileService.scala

```diff
@@ -34,6 +36,39 @@ class FileService(
     } yield Files(files.map(_.fileid))
   }
 
+  def addFile(addFileAndMetadataInput: AddFileAndMetadataInput, userId: UUID): Future[List[FileSequence]] = {
+    val now = Timestamp.from(timeSource.now)
+    val consignmentId = addFileAndMetadataInput.consignmentId
+    val fileRows: List[FileRow] = List.fill(addFileAndMetadataInput.metadataInput.size)(1)
+      .map(_ => {
+        FileRow(uuidSource.uuid, consignmentId, userId, now)
+      })
+    val metadataWithIds: List[(UUID, MetadataInput)] = fileRows.map(_.fileid).zip(addFileAndMetadataInput.metadataInput)
+    val row: (UUID, String, String) => FilemetadataRow = FilemetadataRow(uuidSource.uuid, _, _, now, userId, _)
+
+    val fileMetadataRows: Seq[FilemetadataRow] = metadataWithIds.flatMap {
+      case (fileId, input) =>
+        Seq(
+          row(fileId, input.originalPath.getOrElse(""), ClientSideOriginalFilepath),
+          row(fileId, input.lastModified.toTimestampString, ClientSideFileLastModifiedDate),
+          row(fileId, input.fileSize.map(_.toString).getOrElse(""), ClientSideFileSize),
+          row(fileId, input.checksum.getOrElse(""), SHA256ClientSideChecksum)
+        ) ++ staticMetadataProperties.map(property => {
+          row(fileId, property.value, property.name)
+        })
+      case _ => Seq()
+    }
+    for {
+      _ <- fileRepository.addFiles(fileRows, fileMetadataRows)
+      _ <- if (addFileAndMetadataInput.isComplete) {
+        consignmentStatusRepository.updateConsignmentStatus(consignmentId, "Upload", "Completed", now)
+      } else {
+        Future(List())
```

Comment:
Does this `List()` need to be here? It looks like it means something, but I don't think it's actually being used.

Would `Future(())` work instead?

Reply:
Yeah, `Future(())` works, not sure what I was thinking with the `List()` I've changed it.

Reply:
I wasn't thinking properly yesterday. I think it should be `Future.successful(())` rather than `Future(())`. They result in the same value, but the former returns an immediately completed Future. The latter does some unnecessary asynchronous work.

Reply:
I didn't know that. I've changed it to `Future.successful`